Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+//
+// Authors:
+//   Patrik Torstensson (Patrik.Torstensson@labs2.com)
+//   Wictor Wilén (decode/encode functions) (wictor@ibizkit.se)
+//   Tim Coleman (tim@timcoleman.com)
+//   Gonzalo Paniagua Javier (gonzalo@ximian.com)
+
+//   Marek Habersack <mhabersack@novell.com>
+//
+// (C) 2005-2010 Novell, Inc (http://novell.com/)
+//
+
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Text;
+#if !NO_SYSTEM_WEB_DEPENDENCY && !MOBILE && false
+using System.Web.Configuration;
+#endif
+
+namespace System.Web.Util
+{
+    public
+    class HttpEncoder
+    {
+        static char [] hexChars = "0123456789abcdef".ToCharArray ();
+        static object entitiesLock = new object ();
+        static SortedDictionary <string, char> entities;
+        static Lazy <HttpEncoder> defaultEncoder;
+        static Lazy <HttpEncoder> currentEncoderLazy;
+        static HttpEncoder currentEncoder;
+
+        static IDictionary <string, char> Entities {
+            get {
+                lock (entitiesLock) {
+                    if (entities == null)
+                        InitEntities ();
+
+                    return entities;
+                }
+            }
+        }
+
+        public static HttpEncoder Current {
+            get {
+                if (currentEncoder == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)